tensorflow_addons/layers/normalizations.py

```diff
@@ -321,3 +321,207 @@ def __init__(self, **kwargs):
+        self.epsilon = epsilon
+        self.beta_initializer = tf.keras.initializers.get(beta_initializer)
+        self.gamma_initializer = tf.keras.initializers.get(gamma_initializer)
+        self.beta_regularizer = tf.keras.regularizers.get(beta_regularizer)
+        self.gamma_regularizer = tf.keras.regularizers.get(gamma_regularizer)
+        self.beta_constraint = tf.keras.constraints.get(beta_constraint)
+        self.gamma_constraint = tf.keras.constraints.get(gamma_constraint)
+        self.use_eps_learned = learned_epsilon
+        self.supports_masking = True
+
+        if self.use_eps_learned:
+            self.eps_learned_initializer = tf.keras.initializers.Constant(1e-4)
+        else:
+            self.eps_learned_initializer = None
+
+        if isinstance(axis, list):
+            self.axis = axis[:]
+        elif isinstance(axis, int):
+            self.axis = axis
+        else:
+            raise TypeError("axis must be int or list, type given: %s" % type(axis))
+        self._check_axis()
+
+    def build(self, input_shape):
+        self._check_if_input_shape_is_none(input_shape)
+        self._create_input_spec(input_shape)
+        self._add_gamma_weight(input_shape)
+        self._add_beta_weight(input_shape)
+
+        if self.use_eps_learned:
+            self._add_eps_learned_weight()
+        self.built = True
+        super().build(input_shape)
+
+    def compute_output_shape(self, input_shape):
+        return input_shape
+
+    def get_config(self):
+        config = {
+            "axis": self.axis,
+            "epsilon": self.epsilon,
+            "learned_epsilon": self.use_eps_learned,
+            "beta_initializer": tf.keras.initializers.serialize(self.beta_initializer),
+            "gamma_initializer": tf.keras.initializers.serialize(
+                self.gamma_initializer
+            ),
+            "eps_learned_initializer": tf.keras.initializers.serialize(
+                self.eps_learned_initializer
+            ),
+            "beta_regularizer": tf.keras.regularizers.serialize(self.beta_regularizer),
+            "gamma_regularizer": tf.keras.regularizers.serialize(
+                self.gamma_regularizer
+            ),
+            "beta_constraint": tf.keras.constraints.serialize(self.beta_constraint),
+            "gamma_constraint": tf.keras.constraints.serialize(self.gamma_constraint),
+        }
+        base_config = super().get_config()
+        return dict(**base_config, **config)
+
+    def call(self, inputs):
+        epsilon = tf.math.abs(self.epsilon)
+        if self.use_eps_learned:
+            epsilon += self.eps_learned
+        nu2 = tf.reduce_mean(tf.square(inputs), axis=self.axis, keepdims=True)
+        inputs *= tf.math.rsqrt(nu2 + epsilon)
+        return self.gamma * inputs + self.beta
+
+    def _create_input_spec(self, input_shape):
+        ndims = len(tf.TensorShape(input_shape))
+        for idx, x in enumerate(self.axis):
```

Comment:
`self.axis` can be an `int`. Doing an `enumerate` of `int` will lead to an error:
```python
TypeError: 'int' object is not iterable
```

Reply:
This issue can be fixed with the `isinstance` check in `__init__`.

Reply:
Yes. Thanks

Reply:
We are already checking axis values for integer values which would raise `VlaueError`, hence this check wouldn't be required